bootstrap/tfe_variables.tf

```diff
@@ -46,30 +46,40 @@ locals {
       description = "Personal access token from Github"
       sensitive   = true
       category    = "env"
-    }
+    },
     TFE_ORGANIZATION = {
       value       = var.tfc_organization_name
       description = "Name of the Terraform Cloud organization to manage"
       sensitive   = false
       category    = "env"
-    }
+    },
     TFE_TOKEN = {
-      value       = var.tfc_team_token
+      value       = var.tfc_token
       description = "Terraform token for the 'owners' team."
       sensitive   = true
       category    = "env"
     }
-    vcs_repo_github_app_installation_id = {
-      value       = var.vcs_repo_github_app_installation_id
-      description = "ID for GitHub app installation in TFC. Ensure that the GitHub Terraform app is already installed: https://app.terraform.io/api/v2/github-app/installations."
-      sensitive   = true
-      category    = "terraform"
-    }
-    tfe_organization_name = {
-      value       = var.tfc_organization_name
-      description = "The Terraform Cloud organization to provision resources in (used by Stations station-tfe module)"
-      sensitive   = true
-      category    = "terraform"
-    }
   }
+
+  optional_env_vars = [
+    #You have to set either of them. Both can not be null 
+    var.vcs_repo_github_app_installation_id != null ? {
+      vcs_repo_github_app_installation_id = {
+        value       = var.vcs_repo_github_app_installation_id
+        description = "ID for GitHub app installation in TFC. Ensure that the GitHub Terraform app is already installed: https://app.terraform.io/api/v2/github-app/installations."
+        sensitive   = false
+        category    = "terraform"
+      }
+    } : null,
+    var.vcs_repo_oauth_token_id != null ? {
+      vcs_repo_oauth_token_id = {
+        value       = var.vcs_repo_oauth_token_id
+        description = "ID for GitHub app installation in TFC. Ensure that the GitHub Terraform app is already installed: https://app.terraform.io/api/v2/github-app/installations."
+        sensitive   = false
+        category    = "terraform"
+      }
+    } : null
+  ]
+
+  env_vars = merge(local.base_env_vars, [for env in local.optional_env_vars : env if env != null]...)
```

Comment:
NITPICK
Since optional_env_vars always will contain exactly one element, why not grab the 0 indexed element?

`merge(local.base_env_vars, local.optional_env_vars[0])`

Reply:
Tested this but I could not get it to work. When I changed to the proposed code and run a `terraform plan` the plan output that it want to delete the TFC variable.

```

  # tfe_variable.deployments["vcs_repo_oauth_token_id"] will be destroyed
  # (because key ["vcs_repo_oauth_token_id"] is not in for_each map)
```